internal/repository/clickhouse/aggregators.go

```diff
@@ -40,8 +39,60 @@ func formatWindowSize(windowSize types.WindowSize) string {
 	case types.WindowSizeDay:
 		return "toStartOfDay(timestamp)"
 	default:
-		return "" // No windowing if windowSize is empty or invalid
+		return ""
+	}
+}
+
+func buildFilterConditions(filters map[string][]string) string {
+	if len(filters) == 0 {
+		return ""
+	}
+
+	var conditions []string
+	for key, values := range filters {
+		if len(values) == 0 {
+			continue
+		}
+
+		quotedValues := make([]string, len(values))
+		for i, v := range values {
+			quotedValues[i] = fmt.Sprintf("'%s'", v)
+		}
+
+		conditions = append(conditions, fmt.Sprintf(
+			"JSONExtractString(properties, '%s') IN (%s)",
+			key,
+			strings.Join(quotedValues, ","),
+		))
+	}
+
+	if len(conditions) == 0 {
+		return ""
 	}
+
+	return "AND " + strings.Join(conditions, " AND ")
```

Comment:
NIT: can actually be responsibility of caller to add `AND` or `WHERE` depending on the query strucutre